analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -219,92 +114,74 @@ def get_features(jets, electrons, muons, permutations_dict):
 # - event weighting,
 # - calculating systematic uncertainties at the event and object level,
 # - filling all the information into histograms that get aggregated and ultimately returned to us by `coffea`.
+#
+# #### Machine Learning Task
+#
+# During the processing step, machine learning is used to calculate one of the variables used for this analysis. The models used are trained separately in the `jetassignment_training.ipynb` notebook. Jets in the events are assigned to labels corresponding with their parent partons using a boosted decision tree (BDT). More information about the model and training can be found within that notebook.
 
 # %% tags=[]
-# functions creating systematic variations
-def jet_pt_resolution(pt):
-    # normal distribution with 5% variations, shape matches jets
-    counts = ak.num(pt)
-    pt_flat = ak.flatten(pt)
-    resolution_variation = np.random.normal(np.ones_like(pt_flat), 0.05)
-    return ak.unflatten(resolution_variation, counts)
-
 class TtbarAnalysis(processor.ProcessorABC):
-    def __init__(self,
-                 use_dask,
-                 disable_processing,
-                 io_branches,
-                 ml_options,
-                 xgboost_model_even,
-                 xgboost_model_odd,
-                 permutations_dict={}):
-
-        num_bins = 25
-        bin_low = 50
-        bin_high = 550
-        name = "observable"
-        label = "observable [GeV]"
-        self.hist = (
-            hist.Hist.new.Reg(num_bins, bin_low, bin_high, name=name, label=label)
-            .StrCat(["4j1b", "4j2b"], name="region", label="Region")
-            .StrCat([], name="process", label="Process", growth=True)
-            .StrCat([], name="variation", label="Systematic variation", growth=True)
-            .Weight()
-        )
-
+    def __init__(self, use_dask, use_inference, use_triton):
+
+        # initialize dictionary of hists for signal and control region
+        self.hist_dict = {}
+        for region in ["4j1b", "4j2b"]:
+            self.hist_dict[region] = (
+                hist.Hist.new.Reg(utils.config["global"]["NUM_BINS"], 
+                                  utils.config["global"]["BIN_LOW"], 
+                                  utils.config["global"]["BIN_HIGH"], 
+                                  name="observable", 
+                                  label="observable [GeV]")
+                .StrCat([], name="process", label="Process", growth=True)
+                .StrCat([], name="variation", label="Systematic variation", growth=True)
+                .Weight()
+            )
+        
         self.use_dask = use_dask
```

Comment:
We don't seem to be using this anymore as far as I can see? It was previously controlling the model setup for Triton, how is that working now? Can we remove this property here?